docs/source/credentials.rst

```diff
@@ -0,0 +1,39 @@
+Credentials
+***********
+
+Integration with Git
+====================
+
+Git and DataLad can use each other's credential system.
+In order to allow Git to query credentials from DataLad, the git credential helper delivered with DataLad needs to be configured.
+That is, a section like this needs to be part of one's git config file::
+
+  [credential "https://*.data.example.com"]
+    helper = "datalad"
+
+Note:
+
+- This most likely only makes sense at the user or system level (options `--global`|`--system` with `git config`), since cloning of a repository needs the credentials before there is a local repository.
+- The name of that section is a URL matching expression - see `man gitcredentials`.
+- The URL matching does NOT include the scheme! Hence, if you need to match `http` as well as `https`, you need two such entries.
+- Multiple different git credential helpers can be configured - Git will ask them one after another. For example on OSX, Git comes with a helper to use the OSX keychain which is configured at system level. This does not conflict with setting up DataLad's credential helper.
+
+The other way around, DataLad can ask Git for credentials it knows by other means (other git credential helpers).
+For that a datalad provider config needs to be set up in similar fashion::
+
+  [provider:data_example_provider]
+    url_re = http.*://.*data\.example\.com
+    authentication_type = http_basic_auth
+    credential = data_example_cred
+  [credential:data_example_cred]
+    type = git
```

Comment:
Just to make sure I understand this right, this is where DataLad says "go ask git and see if it has a credential helper for this situation".

If that is the case, I'm pretty convinced that both configurations are needed in order for this to be functional (please correct me if I'm wrong; second time reading through).

It also makes me wonder what kind of weird bugs/corner-cases there could be from having two different regex patterns to match URLs (git's vs Python's). They won't always match exactly the same URLs. Cascading could get very, very weird.

Reply:
I am not sure, where you get the idea from, that both is needed and particularly for the same URL.

However, there probably are cases where a circular setup looks like what one wants. What I can imagine would be a not URL specific config `credential.helper="datalad"` for Git and then encounter a URL for which DataLad is configured to ask Git. In any case: Only makes sense if there's another git-credential-helper that can provide something. That one would need to be configured first, so Git doesn't ask DataLad when it has something else to ask, that only talks to Git.

I will have another look into consequences of circular setups.